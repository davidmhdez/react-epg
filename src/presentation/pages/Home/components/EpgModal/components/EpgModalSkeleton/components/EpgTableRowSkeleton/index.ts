export {default} from "./EpgTableRowSkeleton.tsx"