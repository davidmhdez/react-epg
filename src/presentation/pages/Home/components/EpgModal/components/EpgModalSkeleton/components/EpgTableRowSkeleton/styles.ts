import styled from "styled-components";

export const EpgTableRowSkeletonWrapper = styled.div`
    width: 100%;
    display: flex;
`

export const EpgTableRowChannelColSkeleton = styled.div`
    min-width: 200px;
    height: 110px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    gap: 10px;
    border: 1px solid #6C6C6CAD;
    margin: 5px 10px 5px 0;
    
    .channelNumber{
        width: 60px;
        height: 20px;
    }
    
    .channelImage{
        width: 85px;
        height: 25px;
    }
`