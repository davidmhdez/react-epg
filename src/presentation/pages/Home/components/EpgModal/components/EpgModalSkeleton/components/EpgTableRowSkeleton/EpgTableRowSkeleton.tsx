import React from "react";
import {EpgTableRowChannelColSkeleton, EpgTableRowSkeletonWrapper} from "./styles.ts";
import Skeleton from "react-loading-skeleton";
import EpgTableColSkeleton from "./components/EpgTableColSkeleton";

type Props = {
    programColsWidth: number[]
}

const EpgTableRowSkeleton: React.FC<Props> = ({programColsWidth}) => {
    return (
        <EpgTableRowSkeletonWrapper>
            <EpgTableRowChannelColSkeleton>
                <Skeleton className="channelNumber"/>
                <Skeleton className="channelImage"/>
            </EpgTableRowChannelColSkeleton>
            {programColsWidth.map((col, index)=> (
                <EpgTableColSkeleton key={index} width={col}/>
            ))}
        </EpgTableRowSkeletonWrapper>
    );
};

export default EpgTableRowSkeleton;
