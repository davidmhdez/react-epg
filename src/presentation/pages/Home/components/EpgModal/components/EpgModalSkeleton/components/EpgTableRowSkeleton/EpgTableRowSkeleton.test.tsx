import {render} from "@testing-library/react";
import EpgTableRowSkeleton from "./EpgTableRowSkeleton.tsx";

describe('EpgTableRowSkeleton test', () => {
    test('should match with snapshot', () => {
        const { container } = render(<EpgTableRowSkeleton programColsWidth={[100, 50, 30]}/>);
        expect(container).toMatchSnapshot()
    })
})