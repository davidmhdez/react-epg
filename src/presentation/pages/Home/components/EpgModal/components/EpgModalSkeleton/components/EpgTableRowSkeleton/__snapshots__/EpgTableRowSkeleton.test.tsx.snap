// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgTableRowSkeleton test should match with snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB bvTkNm"
  >
    <div
      class="sc-gEvDqW fGjLCw"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton channelNumber"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton channelImage"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="sc-eqUzNf cIKLZF"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programTitle"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programDuration"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="sc-eqUzNf gCpjOV"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programTitle"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programDuration"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="sc-eqUzNf iAXyfv"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programTitle"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton programDuration"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>
</div>
`;
